tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -602,6 +603,21 @@ DeclResultIdMapper::createFnParam(const ParmVarDecl *param) {
   assert(astDecls[param].instr == nullptr);
   astDecls[param].instr = fnParamInstr;
 
+  if (spirvOptions.debugInfoRich) {
+    // Add DebugLocalVariable information
+    const auto &sm = astContext.getSourceManager();
+    const uint32_t line = sm.getPresumedLineNumber(loc);
+    const uint32_t column = sm.getPresumedColumnNumber(loc);
+    const auto *info = theEmitter.getOrCreateRichDebugInfo(loc);
+    // TODO: replace this with FlagIsLocal enum.
+    uint32_t flags = 1 << 2;
+    auto *debugLocalVar = spvBuilder.createDebugLocalVariable(
+        type, name, info->source, line, column, info->scopeStack.back(), flags,
+        param->getFunctionScopeIndex());
```

Comment:
do we need to add "1" for struct/class member functions?

Reply:
oh I'm not talking about the flag. I'm talking about the `param->getFunctionScopeIndex`. For struct/class member functions, IIRC we create a function whose first parameter is the object itself (`this` param). So I'm wondering if for those cases the parameter index should be shifted by one. We should also have a test for struct/class function parameters.